src/types.ts

```diff
@@ -37,12 +15,105 @@ export interface Spec<T> {
    * A url that leads to more detailed documentation about the env var.
    */
   docs?: string
+  /**
+   * A fallback value, which will be used if the env var wasn't specified. Providing a default effectively makes the env var optional.
+   */
+  default?: NonNullable<T> | undefined
+  /**
+   * A fallback value to use only when NODE_ENV is not 'production'.
+   * This is handy for env vars that are required for production environments, but optional for development and testing.
+   */
+  devDefault?: NonNullable<T> | undefined
+}
+
+export type OptionalSpec<T> = Omit<Spec<T>, 'default'> & { default: undefined }
+export type OptionalTypelessSpec = Omit<OptionalSpec<unknown>, 'choices'>
```

Comment:
Any type that's not used outside this file shouldn't be exported IMO. With the extra exports, it makes it harder to grok which types are actually needed in the module's public interface

Reply:
I've been used to API Extractor in the past and [their strict policy regarding exports](https://api-extractor.com/pages/messages/ae-forgotten-export/). Certainly opinionated! 4f25405bc1e51eb9a13dc7f5c678fcf8e7a5cc03